File: kotlin/Day01.kt
import java.io.File
import java.nio.charset.Charset.defaultCharset

object Day01 {

    fun firstStar(numbers: List<Int>): Int =
        numbers.zipWithNext().count { (x, y) -> y > x }

    fun secondStar(numbers: List<Int>): Int {
        val slidingWindows = numbers.zipWithNext().zip(numbers.drop(2)) { (x, y), z -> x + y + z }
        return firstStar(slidingWindows)
    }
}

fun main() {
    val input = File("resources", "day01.txt").readLines(defaultCharset())
    println(Day01.firstStar(input.map { it.toInt() }))
    println(Day01.secondStar(input.map { it.toInt() }))
}

File: python/Day01.py
def first_star(numbers):
    return sum(1 for (x, y) in list(zip(numbers, numbers[1:])) if y > x)


def second_star(numbers):
    sliding_windows = list(zip(numbers, numbers[1:], numbers[2:]))
    sums = [x + y + z for (x, y, z) in sliding_windows]
    return first_star(sums)


my_input = open('resources/day01.txt', 'r').read()
numbers = list(map(int, my_input.split("\n")))

print(first_star(numbers))
print(second_star(numbers))